transport/tchannel/handler_test.go

```diff
@@ -386,6 +405,22 @@ func TestResponseWriter(t *testing.T) {
 			arg2: []byte(`{"foo":"bar"}` + "\n"),
 			arg3: []byte("{}"),
 		},
+		{
+			format: tchannel.JSON,
+			apply: func(w *responseWriter) {
+				headers := transport.HeadersFromMap(map[string]string{"FoO": "bAr"})
+				w.AddHeaders(headers)
+
+				_, err := w.Write([]byte("{"))
+				require.NoError(t, err)
+
+				_, err = w.Write([]byte("}"))
```

Comment:
why do we need 2 writes here, can we just do `require.NoError(t, io.WriteString(w, "{}"))`